src/app/src/main/kotlin/org/vaccineimpact/api/app/RedirectValidator.kt

```diff
@@ -0,0 +1,24 @@
+package org.vaccineimpact.api.app
+
+import org.vaccineimpact.api.app.errors.BadRequest
+
+open class RedirectValidator
+{
+    @Throws(BadRequest::class)
+    open fun validateRedirectUrl(redirectUrl: String?)
+    {
+        if (redirectUrl == null || redirectUrl.isEmpty()
+                || redirectUrlIsValid(redirectUrl))
+            return
+
+        throw BadRequest("Redirect url domain must be one of http://localhost," +
+                " https://support.montagu.dide.ic.ac.uk, https://montagu.vaccineimpact.org")
```

Comment:
Maybe put these allowed domains in an array, even if it's not configurable, to avoid duplication below and avoid inconsistencies?